echopop/analysis.py

```diff
@@ -243,6 +252,106 @@ def stratified_summary(
     return stratified_results, analysis_dict
 
 
+def variogram_analysis(
+    variogram_parameters: dict,
+    optimization_parameters: dict,
+    transect_dict: dict,
+    settings_dict: dict,
+    isobath_df: pd.DataFrame,
+    fit_parameters: Union[str, List[float]],
+    initial_values: Optional[Union[List[Tuple[str, float]], Dict[str, Dict[str, float]]]],
+    lower_bounds: Union[List[Tuple[str, float]], Dict[str, Dict[str, float]]],
+    upper_bounds: Optional[Union[List[Tuple[str, float]], Dict[str, Dict[str, float]]]],
+):
+
+    # Extract specific variogram parameters
+    # ---- Number of lags
+    n_lags = variogram_parameters["n_lags"]
+    # ---- Lag resolution
+    lag_resolution = variogram_parameters["lag_resolution"]
+
+    # Compute the lag distances
+    distance_lags = np.arange(1, n_lags) * lag_resolution
+    # ---- Add to the `variogram_parameters` dictionary
+    variogram_parameters["distance_lags"] = distance_lags
+    # ---- Update the max range parameter, if necessary
+    max_range = lag_resolution * n_lags
+    # ---- Mismatched value in default parameters should be replaced
+    if "range" in variogram_parameters and variogram_parameters["range"] != max_range:
+        # ---- Update
+        variogram_parameters["range"] = max_range
+        # ---- Skip file dir in message
+        _warn_skips = (os.path.dirname(__file__),)
+        # ---- Generate warning
+        warnings.warn(
+            f"Default `range` parameter ({variogram_parameters['range']}) does not "
+            f"align with the inputs for `lag_resolution` ({lag_resolution}) "
+            f"and `n_lags` ({n_lags}). The range will be changed to {max_range}.",
+            stacklevel=2,
+            skip_file_prefixes=_warn_skips,
+        )
```

Comment:
I am confused by this. From the if statement, it seems that unless `variogram_parameters["range"]` is set to `max_range`, it will be reset to `max_range` and a warning will be raised. Wouldn't this be the same as not having `range` in the `variogram_parameters` dictionary  and just compute it from `n_lag * lag_resolution`? Did I miss something here?